setup/setup.iss

```diff
@@ -362,16 +369,33 @@ begin
         if not CreateSymbolicLinkA(GetGameDir('RedFaction.exe'), ExpandConstant('{app}\DashFactionLauncher.exe'), 0) then
         begin
             Log('CreateSymbolicLink failed');
-            MsgBox('Failed to replace Red Faction launcher by a symbolic link to Dash Faction launcher.', mbError, MB_OK);
+            MsgBox('Failed to replace the Red Faction launcher with a symbolic link to the Dash Faction launcher.', mbError, MB_OK);
+        end;
+    end;
+end;
+
+procedure RestoreRedFactionLauncher;
+var 
+   GamePath: String;
+begin
+    // Use the registry entry to find the game directory
+    RegQueryStringValue(HKCU, 'Software\Volition\Red Faction\Dash Faction', 'Executable Path', GamePath);
+    GamePath := ExtractFileDir(GamePath) + '\';
+    if FileExists(GamePath + 'RedFaction.exe.bak') then

```

Comment:
hm, I wonder if it's possible to check if launcher was replaced during the installation process, because now we may delete a proper file just because someone messed with his RF directory. But I guess it's not a big deal and can fix the game for some people so I approve this change

Reply:
But it is totally possible that someone could change the game exectuable path to something else and then uninstall the launcher, and the files would not be restored. I will explore other options.

Reply:
Perhaps InnoSetup already does it... @Mystyled could you check if `WizardSelectedTasks` returns tasks selected during the installation? Also if does not work `WizardSelectedComponents` is also worth a try if we change task into a component.
My worry is not about someone changing exe path after installation but about unconditionally using RF.exe.bak file without any verification that this file was created by the installer and not by the user. User could install DF without replace task enabled and have some bak file in RF directory that he created himself (it can be anything, e.g. RF 1.00, some RF.exe from FF, even PF with changed name) and may be unaware that uninstaller will replace exe with that bak file without asking. It would be better if it was done only if user previously asked for replacement. Another option would be to ask the user if he wants RF.exe restored.

Reply:
`WizardSelectedTasks` does return all the selected tasks as expected. We could check the uninstall info in the registry to see if `replacerflauncher` was ticked during installation before we restore the launcher.

Reply:
`replacerflauncher` is a task so if `WizardSelectedTasks` (and `WizardIsTaskSelected`) works in uninstaller in the way that it returns tasks that were active during the installation, that's all we need. No need to directly access registry keys. Am I missing something?

Reply:
I tried using those functions during uninstall and got an error for each. Apparently they cannot be called in `[Code]` during uninstallation: https://stackoverflow.com/questions/28591404/uninstaller-checking-if-a-task-was-selected-during-the-installation

Reply:
I finally got this to work. The whole process:

1. Right before uninstallation, look at the uninstall registry entry and check if `replacerflauncher` was ticked during installation
2. Use the executable path entered in the launcher to get the game directory
3. Check if a backup of `RedFaction.exe` exists
4. Do the restoration

Another precaution that could be taken could be to store the symlink path in a registry value separate from the game executable and use that as @GooberRF suggested, but then that could introduce problems if the game folder were to be moved to a separate location and the value isn't updated.

![image](https://user-images.githubusercontent.com/89315313/180875438-17398d25-a6cc-45c2-8d30-8a6e5c027b6d.png)

Reply:
Sounds good, I don't think we need more precautions. 
BTW I think your code in the screenshot is over-complicated. You don't need ExpandConstant for AppId retrival according to https://stackoverflow.com/questions/20021317/how-to-convert-the-contents-of-appid-in-setup-into-a-string-in-code I think you can just embed preprocessor expression in the string

Reply:
For some reason, the code fails to work if I don't use it. I have tried these:

`'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\' + '{#SetupSetting("AppId")}' + '_is1'`
`'SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{#SetupSetting("AppId")}_is1'`

Reply:
Have you tried displaying them? Maybe something changed between inno setup 5 and 6 in this regard...  I would have expected them to work..

Reply:
They do display correctly in the output. But now I know what the problem is.

`SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{{BDD60DE7-9374-463C-8E74-8227EB03E28F}_is1`

Without ExpandConstant, the extra '{' at the beginning of the AppId is kept and the key is not found.

Reply:
It seems `ExpandConstant` is the proper way to handle it: https://stackoverflow.com/a/58945606